pkg/cache/cache.go

```diff
@@ -368,7 +375,7 @@ func (c *Cache) cleanupAssumedState(w *kueue.QueuedWorkload) {
 	if assumed {
 		// If the workload's assigned ClusterQueue is different from the assumed
 		// one, then we should also cleanup the assumed one.
-		if assumedCapName != string(w.Spec.AssignedClusterQueue) {
+		if w.Spec.Admission != nil && assumedCapName != string(w.Spec.Admission.ClusterQueue) {
```

Comment:
This is not exactly the same as before, but I believe this should be the correct implementation.

But the change forced me to change the "update assumed workload" to use AddOrUpdate instead of Update, which is actually what the event handler does.